src/main/java/thePackmaster/actions/quietpack/FrontflipAction.java

```diff
@@ -0,0 +1,46 @@
+package thePackmaster.actions.quietpack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.GameActionManager;
+import com.megacrit.cardcrawl.actions.common.DrawCardAction;
+import com.megacrit.cardcrawl.actions.utility.WaitAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.vfx.PlayerTurnEffect;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import static thePackmaster.util.Wiz.adp;
+
+public class FrontflipAction extends AbstractGameAction {
```

Comment:
Personally I'd replace this entire action with an anonymous action that also uses the DiscardSpecificCardAction instead of doing it manually.